test/conformance/generatename_test.go

```diff
@@ -0,0 +1,187 @@
+	}
+}
+
+func setConfigurationGenerateName(generateName string) rtesting.ConfigOption {
+	return func(config *v1alpha1.Configuration) {
+		config.ObjectMeta.GenerateName = generateName
+	}
+}
+
+func setRouteGenerateName(generateName string) rtesting.RouteOption {
+	return func(route *v1alpha1.Route) {
+		route.ObjectMeta.GenerateName = generateName
+	}
+}
+
+// getGenerateNamePrefix returns the object name to be used for testing, shorted to
+// 44 characters to avoid #3236, as generateNames longer than 44 characters may cause
+// some knative resources to never become ready.
+func getGenerateNamePrefix(t *testing.T) string {
+	generateName := test.ObjectNameForTest(t) + "-"
+
+	generateNameLength := len(generateName)
+	if generateNameLength > 44 {
+		generateNameLength = 44
+	}
+	return generateName[0:generateNameLength]
+}
+
+func ensureCanServeRequests(t *testing.T, clients *test.Clients, route *v1alpha1.Route) {
+	t.Logf("Route %s has a domain set in its status", route.Name)
+	var domain string
+	err := test.WaitForRouteState(
+		clients.ServingClient,
+		route.Name,
+		func(r *v1alpha1.Route) (bool, error) {
+			domain = r.Status.Domain
+			return domain != "", nil
+		},
+		"RouteDomain",
+	)
+
+	t.Logf("Route %s can serve the expected data at the endpoint", route.Name)
+	_, err = pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		t.Logf,
+		domain,
+		pkgTest.Retrying(pkgTest.EventuallyMatchesBody(helloWorldText), http.StatusNotFound),
+		"WaitForEndpointToServeText",
+		test.ServingFlags.ResolvableDomain)
+	if err != nil {
+		t.Fatalf("The endpoint for Route %s at domain %s didn't serve the expected text \"%s\": %v", route.Name, domain, helloWorldText, err)
+	}
+
+}
+
+// TestServiceGenerateName checks that knative Services MAY request names generated by
+// the system using metadata.generateName. It ensures that knative Services created this way can become ready
+// and serve requests.
+func TestServiceGenerateName(t *testing.T) {
+	t.Parallel()
+	clients := setup(t)
+
+	generateName := getGenerateNamePrefix(t)
+	names := test.ResourceNames{
+		Image: helloworld,
+	}
+
+	// Cleanup on test failure.
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+	defer func() { test.TearDown(clients, names) }()
+
+	// Create the service using the generate name field. If the serivce does not become ready this will fail.
+	t.Logf("Creating new service with generateName %s", generateName)
+	resources, err := test.CreateRunLatestServiceReady(t, clients, &names, &test.Options{}, setServiceGenerateName(generateName))
+	if err != nil {
+		t.Fatalf("Failed to create service with generateName %s: %v", generateName, err)
+	}
+
+	// Ensure that the name given to the service is generated from the generateName field.
+	t.Log("When the service is created, the name is generated using the provided generateName")
+	if names.Service == generateName {
+		t.Fatalf("Service did not recieve a randomized name, instead got %s", names.Service)
```

Comment:
I would recommend t.Errorf for this and below as this failure does not prohibit further validation.

Reply:
Fixed.